### DISCUSSION TITLE
Grpc-Gateway and ConnectClient

### DISCUSSION THREAD

**Discussion by justmumu** [connectrpc/connect-go#545] [Q&A] [answered]
Hi all,

Is there a way to use connect client in grpc-gateway instead of grpc.Dial ?

I am trying to integrate open telemetry tracing using github.com/bufbuild/connect-opentelemetry-go. The library expects to be incoming requests made by connect generated clients to use same traceID.

**Comment by akshayjshah** [comment, marked as answer]
I don't believe that there is, but that's a better question to ask in the grpc-gateway repository :)

We're working on a bit of middleware that should be able to replace grpc-gateway in many cases, though - stay tuned!

**Reply by justmumu** [reply]
Hi Akshay,

I would be happy to contribute to something like this.

If you think, you can guide me about what you need.

Thanks.

**Reply by emcfarlane** [reply]
@justmumu we've just released the open source project [vanguard](https://github.com/connectrpc/vanguard-go) for transcoding RPC and REST protocols. In many cases this can be used like gRPC-Gateway. Wrap in front of your server's handler to transcode REST clients or convert protocols (Connect->gRPC). Have a look through the docs and reach out if you have any issues!

 https://pkg.go.dev/connectrpc.com/vanguard#example-package-RestClientToRpcServer
